code/controllers/subsystem/overlays.dm

```diff
@@ -79,6 +79,13 @@ SUBSYSTEM_DEF(overlays)
 		if(!overlay)
 			continue
 		if (istext(overlay))
+#ifdef UNIT_TESTS
+			// This is too expensive to run normally but running it during CI is a good test
+			var/list/icon_states_available = icon_states(icon)
+			if(!(overlay in icon_states_available))
+				stack_trace("Invalid icon state '[overlay]' attempted to get added as an overlay to [src].")
+				continue
```

Comment:
Would it be possible to list the icon file in use? it'd reduce the amount of clicking around needed to solve issues.

Might need a case for the autogenerated stuff tho, idk

Reply:
Typepath instead of var name would also be nice, but that might just be a me thing

Reply:
I'll see what I can do with the first and with the second I'll include both, even in this pr the name helped me figure out an icon issue resulting from a map edit because the map edited version was a different name.